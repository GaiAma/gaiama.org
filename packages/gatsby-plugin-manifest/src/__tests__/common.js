const path = require(`path`)
const { defaultIcons, doesIconExist, addDigestToPath } = require(`../common`)

describe(`gatsby-plugin-manifest`, () => {
  describe(`defaultIcons`, () => {
    it(`includes all icon sizes`, () => {
      expect(defaultIcons).toMatchSnapshot()
    })
  })

  describe(`doesIconExist`, () => {
    it(`returns true if image exists on filesystem`, () => {
      const iconSrc = path.resolve(__dirname, `./images/gatsby-logo.png`)
      expect(doesIconExist(iconSrc)).toBeTruthy()
    })

    it(`returns false if image does not exist on filesystem`, () => {
      const iconSrc = path.resolve(__dirname, `./images/non-existent-logo.png`)
      expect(doesIconExist(iconSrc)).toBeFalsy()
    })
  })

  describe(`addDigestToPath`, () => {
    it(`returns unmodified path`, () => {
      expect(
        addDigestToPath(`icons/icon-48x48.png`, `thisismydigest`, `none`)
      ).toBe(`icons/icon-48x48.png`)
    })

    it(`returns query modified path`, () => {
      expect(
        addDigestToPath(`icons/icon-48x48.png`, `thisismydigest`, `query`)
      ).toBe(`icons/icon-48x48.png?v=thisismydigest`)
    })

    it(`returns fileName modified path`, () => {
      expect(
        addDigestToPath(`icons/icon-48x48.png`, `thisismydigest`, `name`)
      ).toBe(`icons/icon-48x48-thisismydigest.png`)
    })
  })
})
