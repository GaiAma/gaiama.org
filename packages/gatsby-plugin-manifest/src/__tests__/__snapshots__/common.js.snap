// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest defaultIcons includes all icon sizes 1`] = `
Array [
  Object {
    "sizes": "48x48",
    "src": "icons/icon-48x48.png",
    "type": "image/png",
  },
  Object {
    "sizes": "72x72",
    "src": "icons/icon-72x72.png",
    "type": "image/png",
  },
  Object {
    "sizes": "96x96",
    "src": "icons/icon-96x96.png",
    "type": "image/png",
  },
  Object {
    "sizes": "144x144",
    "src": "icons/icon-144x144.png",
    "type": "image/png",
  },
  Object {
    "sizes": "192x192",
    "src": "icons/icon-192x192.png",
    "type": "image/png",
  },
  Object {
    "sizes": "256x256",
    "src": "icons/icon-256x256.png",
    "type": "image/png",
  },
  Object {
    "sizes": "384x384",
    "src": "icons/icon-384x384.png",
    "type": "image/png",
  },
  Object {
    "sizes": "512x512",
    "src": "icons/icon-512x512.png",
    "type": "image/png",
  },
]
`;
